src/modules/main/navbar/user-section/user-details/UserDetails.tsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+
+import { Avatar } from '@material-ui/core';
+
+import { Button, Img } from 'ui';
+
+import { TemplateCategory } from 'core/api';
+import { useAuthProvider } from 'core/auth';
+
+import csx from './UserDetails.scss';
+
+const MAX_VISIBLE_TECHNOLOGIES = 16;
+
+const UserDetails = () => {
+  const {
+    user: {
+      username,
+      email,
+      company,
+      seniority,
+      technologies,
+      yearsOfExperience,
+      position,
+      connectedWithGithub
+    },
+    logOut
+  } = useAuthProvider();
+
+  const { replace } = useHistory();
+
+  const DETAILS: Record<string, string | number> = {
+    Company: company || 'Unknown',
+    'Years of experience': yearsOfExperience || 'Unknown',
+    Seniority: seniority || 'Unknown',
+    Position: position || 'Unknown'
+  };
+
+  return (
+    <div className={csx.userDetails}>
+      <div className={csx.header}>
+        <Avatar className={csx.avatar}>{username.charAt(0).toUpperCase()}</Avatar>
+        <div className={csx.headerDetails}>
+          <span>{username}</span>
+          <span>{email}</span>
+        </div>
+      </div>
+
+      <div className={csx.details}>
+        {Object.keys(DETAILS).map((detail, idx) => {
+          const value = Object.values(DETAILS)[idx];
+
+          return (
+            <div className={csx.detail} key={idx}>
+              <span>{detail}</span>
+              <span>{`${value} ${
+                detail === 'Years of experience' && value != 'Unknown' ? 'years' : ''
+              }`}</span>
+            </div>
+          );
+        })}
+
+        {technologies.length > 0 ? (
+          <div className={csx.detail}>
+            <span>{`Technologies ${
+              technologies.length > MAX_VISIBLE_TECHNOLOGIES ? `(${technologies.length})` : ''
+            }`}</span>
+
+            <div className={csx.technologies}>
+              {technologies.map((technology, idx) => {
+                const { name, pictureUrl } = technology;
```

Comment:
Should be a function which will be memoized. Adding computations and logic inside tsx / jsx decreases readability.